src/main/java/org/wildfly/naming/client/AbstractContext.java

```diff
@@ -332,12 +332,8 @@ Name decomposeName(CompositeName compositeName) throws NamingException {
             return new SimpleName();
         }
         final NameParser parser = getNativeNameParser();
-        Name name = parser.parse(compositeName.get(0));
-        final int size = compositeName.size();
-        if (size == 1) {
-            return name;
-        }
-        for (int i = 1; i < size; i ++) {
+        final Name name = new SimpleName();
```

Comment:
I think this should use the context's naming parser, because it might not be compatible with SimpleName.